app/controllers/api/v4/call_results_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Api::V4::CallResultsController < Api::V4::SyncController
+  def sync_from_user
+    __sync_from_user__(call_result_params)
+  end
+
+  private
+
+  def merge_if_valid(call_result_params)
+    validator = Api::V4::CallResultPayloadValidator.new(call_result_params)
+    logger.debug "CallResult had errors: #{validator.errors_hash}" if validator.invalid?
+    if validator.check_invalid?
+      {errors_hash: validator.errors_hash}
+    else
+      transformed_params = Api::V4::Transformer.from_request(call_result_params)
+      call_result = CallResult.merge(transformed_params)
+
+      {record: call_result}
+    end
+  end
+
+  def call_result_params
+    params.require(:call_results).map do |call_result_params|
+      call_result_params.permit(
+        :id,
+        :user_id,
+        :appointment_id,
+        :cancel_reason,
+        :result,
+        :deleted_at,
+        :created_at,
+        :updated_at
```

Comment:
We do the automagic to shove these two timestamps into `device_*` fields, right?

Reply:
Yes, this is done by the call to `Api::V4::Transformer.from_request` in `merge_if_valid`